common/oatbox/user/UserLanguageService.php

```diff
@@ -64,7 +64,8 @@ public function getInterfaceLanguage(User $user)
         }
 
         $lang = $user->getPropertyValues(GenerisRdf::PROPERTY_USER_UILG);
-        return empty($lang) ? $this->getDefaultLanguage() : (string)current($lang);
+
+        return empty($lang) ? $this->getOption(self::OPTION_INTERFACE_LANGUAGE, DEFAULT_LANG) : (string)current($lang);
```

Comment:
- [ ] Please update unit tests

Reply:
Here they are :point_right: [UserLanguageServiceTest.php](https://github.com/oat-sa/generis/blob/876a8d453f0b1f95ffd03db8ebc9305e855d8447/test/unit/oatbox/user/UserLanguageServiceTest.php)

They cover:
- User with UI lang, `OPTION_INTERFACE_LANGUAGE` set --> Returns user UI lang
- User with UI lang set, `OPTION_INTERFACE_LANGUAGE` **not** set --> Returns user UI lang
- User with **no** UI lang set, `OPTION_INTERFACE_LANGUAGE` set --> Returns `OPTION_INTERFACE_LANGUAGE`
- User with **no** UI lang set, `OPTION_INTERFACE_LANGUAGE` **not** set --> Returns `DEFAULT_LANG`

I've checked all branches from this method are reported as covered by PHPUnit if put this way:

```
public function getInterfaceLanguage(User $user)
{
    if (!empty($this->customInterfaceLanguage)) {
        return $this->customInterfaceLanguage;
    }

    $lang = $user->getPropertyValues(GenerisRdf::PROPERTY_USER_UILG);

    // return empty($lang) ? $this->getOption(self::OPTION_INTERFACE_LANGUAGE, DEFAULT_LANG) : (string)current($lang);

    if (empty($lang)) {
        return $this->getOption(self::OPTION_INTERFACE_LANGUAGE, DEFAULT_LANG);
    }

    return (string)current($lang);
}
```

Reply:
I'm marking this thread as solved since all branches from this method are covered by unit tests.